From f818b0e62d2112bcda9070341ce8c8ef9e7065c7 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Fri, 20 Sep 2019 11:51:06 -0400
Subject: [PATCH 12/26] oxt: flask: Add stubdom_pasthrough_dev interface

This bundles up all the required permissions for PCI passhrough to a
domain with or without a stubdom.

We default to requiring an IOMMU, but not requiring interrupt remapping.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 tools/flask/policy/modules/xen.if | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/tools/flask/policy/modules/xen.if b/tools/flask/policy/modules/xen.if
index 1116fdaeea..121ab2ab48 100644
--- a/tools/flask/policy/modules/xen.if
+++ b/tools/flask/policy/modules/xen.if
@@ -246,6 +246,19 @@ define(`dm_delegate_devices', `
     delegate_devices($2, $3_target)
 ')
 
+# stubdom_passthrough_dev(priv-domain, dm-domain, target-domain, device)
+#   Allow devices to be delegated through a stubdom and used by the target
+define(`stubdom_passthrough_device',`
+    # These are additionally required for non-stubdom passthrough
+    admin_device($1, $4)
+    delegate_devices($1, $3)
+    use_device_iommu_nointremap($3, $4)
+    # These are additionally required for Stubdom passthrough
+    admin_device($2, $4)
+    dm_delegate_devices($1, $2, $3)
+    use_device_iommu_nointremap($2, $4)
+')
+
 # argo_comms(domA, domB)
 #   Allow two domains to talk argo to each other
 define(`argo_comms', `
-- 
2.21.0

